src/poly.rs

```diff
@@ -128,6 +131,50 @@ impl<F, B> Polynomial<F, B> {
     }
 }
 
+pub(crate) fn batch_invert_assigned<F: FieldExt>(
+    assigned: &[Polynomial<Assigned<F>, LagrangeCoeff>],
+) -> Vec<Polynomial<F, LagrangeCoeff>> {
+    let mut assigned_denominators: Vec<_> = assigned
+        .iter()
+        .map(|f| {
+            f.iter()
+                .map(|value| value.denominator())
+                .collect::<Vec<_>>()
+        })
+        .collect();
+
+    assigned_denominators
+        .iter_mut()
+        .flat_map(|f| {
+            f.iter_mut()
+                // If the denominator is zero or one, we can skip it, reducing the
+                // size of the batch inversion.
+                .filter(|d| !(d.is_zero() || **d == F::one()))
+        })
+        .batch_invert();
+
+    assigned
+        .iter()
+        .zip(assigned_denominators.into_iter())
+        .map(|(poly, inv_denoms)| poly.invert(inv_denoms))
+        .collect()
+}
+
+impl<F: Field> Polynomial<Assigned<F>, LagrangeCoeff> {
+    pub(crate) fn invert(&self, inv_denoms: Vec<F>) -> Polynomial<F, LagrangeCoeff> {
```

Comment:
This is actually doing a Hadamard product, which could be useful for other things, I guess.

Reply:
I've kept it internal for now, as I want to compute a single batch inversion with all advice or fixed columns, instead of per-column, which means the API is not nicely encapsulated.